watchtower/lookout/justice_descriptor.go

```diff
@@ -166,42 +167,72 @@ func (p *JusticeDescriptor) assembleJusticeTxn(txWeight int64,
 		})
 	}
 
-	// Using the total input amount and the transaction's weight, compute
-	// the sweep and reward amounts. This corresponds to the amount returned
-	// to the victim and the amount paid to the tower, respectively. To do
-	// so, the required transaction fee is subtracted from the total, and
-	// the remaining amount is divided according to the prenegotiated reward
-	// rate from the client's session info.
-	sweepAmt, rewardAmt, err := p.SessionInfo.ComputeSweepOutputs(
-		totalAmt, txWeight,
-	)
-	if err != nil {
-		return nil, err
-	}
+	policy := p.SessionInfo.Policy
+	if policy.BlobType.Has(blob.FlagReward) {
+		// Using the total input amount and the transaction's weight,
+		// compute the sweep and reward amounts. This corresponds to the
+		// amount returned to the victim and the amount paid to the
+		// tower, respectively. To do so, the required transaction fee
+		// is subtracted from the total, and the remaining amount is
+		// divided according to the prenegotiated reward rate from the
+		// client's session info.
+		sweepAmt, rewardAmt, err := policy.ComputeRewardOutputs(
+			totalAmt, txWeight,
+		)
+		if err != nil {
+			return nil, err
+		}
 
-	// TODO(conner): abort/don't add if outputs are dusty
+		// Add the sweep and reward outputs to the justice transaction.
+		justiceTxn.AddTxOut(&wire.TxOut{
+			PkScript: p.JusticeKit.SweepAddress[:],
```

Comment:
IMO we should eventually re-name the justice kit elements to `*Script` instead of `*Address`.

Reply:
Agreed, i will add it to the list of todos